src/Domain/Infra/InMemory/DomainEntityRepositoryTrait.php

```diff
@@ -126,6 +124,66 @@ private function createResultSet(iterable $entities, int $offset = 0, int $limit
         return new DomainCollection($entities);
     }
 
+    private function isValidEntityId(array $ids): bool
+    {
+        if (count($ids) !== count($this->idFields)) {
+            return false;
+        }
+
+        foreach ($ids as $id) {
+            if (null === $this->normalizeEntityId($id)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function normalizeEntityId($id)
+    {
+        if ($id instanceof DomainIdInterface) {
+            return $id->isEmpty() ? null : $id;
+        } elseif (is_object($id)) {
```

Comment:
The checks here are all mutually exclusive and they have returns, so they can act as guard clauses instead of elseif